game/src/character/objects/TreasureChest.java

```diff
@@ -0,0 +1,85 @@
+package character.objects;
+
+import basiselements.AnimatableElement;
+import character.player.Hero;
+import collision.CharacterDirection;
+import collision.Collidable;
+import collision.Hitbox;
+import graphic.Animation;
+import java.util.ArrayList;
+import java.util.List;
+import level.elements.ILevel;
+import textures.TextureHandler;
+import tools.Point;
+
+public class TreasureChest extends AnimatableElement implements Collidable {
+    private Animation closed;
+    private Animation opening;
+    private Animation opened;
+    private Animation currentAnimation;
+    private Point currentPosition;
+    private ILevel currentLevel;
+    private boolean isOpen = false;
+    private int counter = 15;
+    protected Hitbox hitbox;
+    private List<Item> inventory;
+
+    public TreasureChest() {
+        List<String> texturePaths = TextureHandler.getInstance().getTexturePaths("ui_heart_full");
+        closed = new Animation(texturePaths, 1);
+        texturePaths = TextureHandler.getInstance().getTexturePaths("ui_heart");
+        opening = new Animation(texturePaths, 5);
+        texturePaths = TextureHandler.getInstance().getTexturePaths("ui_heart_empty");
+        opened = new Animation(texturePaths, 1);
+
+        currentAnimation = closed;
+
+        hitbox = new Hitbox(6, 6);
+        hitbox.setCollidable(this);
+        inventory = new ArrayList<>();
+    }
+
+    public void setLevel(ILevel level) {
```

Comment:
Hier würde ich das anders machen.
Im Konstruktor den `Point currentPosition` mitgeben und dort das lokale Attribut setzten. 
Dann braucht es diese Methode nicht mehr, und die Kiste braucht das Level nicht zu kenne. 
In `Starter` würde ich dann die Position mit `levelAPI.getCurrentLevel().getStartTile().getCoordinate().toPoint()` der Kiste übergeben.